File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 7/10/2025, 11:58:22 AM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        buy_price = prices[0]
        for x in prices[1:]:
            if buy_price > x:
                buy_price = x
            z = x - buy_price
            profit = max(profit, z)
        return profit

                

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.c
// Last updated: 7/10/2025, 11:58:21 AM
int maxProfit(int* prices, int pricesSize) {
    int buyDay = 0;
    int profit = 0;
    int temp_profit;
    for (int i = 1; i < pricesSize; i++){
        temp_profit = prices[i] - prices[buyDay];
        if (prices[i] < prices[buyDay]){
            buyDay = i;
        } else if (temp_profit > profit){
            profit = temp_profit;
        }
    }
    return profit;
}